File: packages/webservice/models/src/main/kotlin/com/yahoo/navi/ws/models/beans/fragments/DeliveryFormat.kt
/**
 * Copyright 2021, Yahoo Holdings Inc.
 * Licensed under the terms of the MIT license. See accompanying LICENSE.md file for terms.
 */
package com.yahoo.navi.ws.models.beans.fragments

import com.fasterxml.jackson.annotation.JsonInclude
import com.yahoo.navi.ws.models.beans.enums.DeliveryType
import com.yahoo.navi.ws.models.types.JsonType
import org.hibernate.annotations.TypeDef

@JsonInclude(JsonInclude.Include.NON_NULL)
@TypeDef(typeClass = JsonType::class, name = "json")
data class DeliveryFormat(
    var type: DeliveryType? = null
)

File: packages/core/addon/models/fragments/delivery-format.ts
/**
 * Copyright 2022, Yahoo Holdings Inc.
 * Licensed under the terms of the MIT license. See accompanying LICENSE.md file for terms.
 */
import { attr } from '@ember-data/model';
import Fragment from 'ember-data-model-fragments/fragment';
import type { DeliveryFormatType } from 'navi-core/models/registry';

export default class DeliveryFormatFragment extends Fragment {
  @attr('string')
  declare type: string;

  @attr()
  declare options: unknown;
}

export interface TypedDeliveryFormat extends DeliveryFormatFragment {
  type: DeliveryFormatType;
}
